### PR TITLE
EPMRPP-38676 || Pattern analysis impl model

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/CreatePatternTemplateRQ.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import static com.epam.ta.reportportal.ws.model.ValidationConstraints.MIN_ANALYSIS_PATTERN_NAME_LENGTH;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CreatePatternTemplateRQ {
+
+	@NotNull
+	@NotEmpty
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/CreatePatternTemplateRQ.java]: `@NotEmpty` checks for null, so you can remove `@NotNull`

File: src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/CreatePatternTemplateRQ.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	@Size(min = MIN_ANALYSIS_PATTERN_NAME_LENGTH, max = MAX_ANALYSIS_PATTERN_NAME_LENGTH)
+	@JsonProperty(value = "name")
+	private String name;
+
+	@NotNull
+	@NotEmpty
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/CreatePatternTemplateRQ.java]: `@NotEmpty` checks for null, so you can remove `@NotNull`

File: src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/CreatePatternTemplateRQ.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	@JsonProperty(value = "value")
+	private String value;
+
+	@NotNull
+	@NotEmpty
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/CreatePatternTemplateRQ.java]: `@NotEmpty` checks for null, so you can remove `@NotNull`

File: src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/UpdatePatternTemplateRQ.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import static com.epam.ta.reportportal.ws.model.ValidationConstraints.MIN_ANALYSIS_PATTERN_NAME_LENGTH;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class UpdatePatternTemplateRQ {
+
+	@NotNull
+	@NotEmpty
```
Comment (ihar-kahadouski) [inline src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/UpdatePatternTemplateRQ.java]: `@NotEmpty` checks for null, so you can remove `@NotNull`

File: src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/CreatePatternTemplateRQ.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/ws/model/project/config/pattern/CreatePatternTemplateRQ.java]: NotEmptyValidator allows the value to be 'null', but only if it's 'not null' validation constraint will check whether the String is empty or not

**Pull request merged**